department-of-veterans-affairs/caseflow
# Remove attorney capability to change overtime

Resolves #13584 

### Description
When feature toggle `:overtime_revamp` is enabled, Attorneys will no longer have the ability to designate OT cases.

### Other approach considered
Considered updating the `CaseReviewsController` but `overtime` was well encapsulated within `params`, which is passed along to `AttorneyCaseReview` and `create` https://github.com/department-of-veterans-affairs/caseflow/blob/43562ed9e75b14c6f949802521da9df6a4214c2b/app/models/attorney_case_review.rb#L72
`params.overtime` is never examined, except to check for permitted params https://github.com/department-of-veterans-affairs/caseflow/blob/4b7ca54f05d592ef7d635d86463cefa2f12a92d1/app/controllers/case_reviews_controller.rb#L53
So I decided to keep `overtime` hidden and well-encapsulated within `params`.

The code change was straight-forward in the front-end code: use the current overtime value provided by `appeal.overtime` as the value to pass to the controller.

I considered several places to add tests (such as `spec/feature/queue/overtime_spec.rb`), and ultimately decided to reuse a lot of the setup and test steps provided by an existing test https://github.com/department-of-veterans-affairs/caseflow/blob/c99b229d857d4787b9b7a0f9fe7bfad0dd87ac1c/spec/feature/queue/ama_queue_spec.rb#L408

### Acceptance Criteria
- [x] This work is behind the feature toggle: :overtime_revamp
- [x] The overtime checkbox should not be accessible to the following user groups: attorneys
- [ ] Update documentation: notify training for documentation updates
- [x] Attorneys will no longer have the ability to designate OT cases

### Testing Plan
0. Enable `overtime_revamp` (`FeatureToggle.enable!(:overtime_revamp)`)
1. Log in as a judge
2. Assign a Legacy case to attorney
   * (try an AMA case)
3. Mark case at Overtime 
   * (try switching the order of 3 and 2)
   * (try not marking the case as overtime)
4. Log in as the attorney to which you assigned the case
5. Go to the Case Details page of that case. Confirm the OT badge exists, if it was marked as OT by the judge (in step 3).
6. In task dropdown, select "Decision ready for review" and go through the process up to the last step (where you fill in the document ID -- see screenshots below). Note that the "This work product is overtime" checkbox is not present.
   * (try disabling `overtime_revamp` (`FeatureToggle.disable!(:overtime_revamp)`), refresh page and go through the process again, and make sure the checkbox appears)
7. Click Continue and a success banner should appear
8. In the Rails console, check the overtime value (`appeal.latest_attorney_case_review.overtime`) should be what the judge marked.

### User Facing Changes

 BEFORE|AFTER
 ---|---
<img width="885" alt="image" src="https://user-images.githubusercontent.com/55255674/83075756-fad40380-a039-11ea-87f8-3fc6afc9c3cc.png"> | <img width="862" alt="image" src="https://user-images.githubusercontent.com/55255674/83075492-92852200-a039-11ea-9d9c-baa2bec93846.png">
